homeassistant/components/weather/buienradar.py

```diff
@@ -79,17 +110,33 @@ def name(self):
 
     @property
     def condition(self):
-        """Return the name of the sensor."""
-        return self._data.condition
+        """Return the current condition."""
+        from buienradar.buienradar import (CONDCODE)
+        try:
+            if self._data and self._data.condition:
+                return self.hass.data[DATA_CONDITION][
+                    self._data.condition[CONDCODE]]
+            return STATE_UNKNOWN
+        except (ValueError, IndexError):
```

Comment:
DATA_CONDITION is a dictionary. Dictionaries raise `KeyError` if the key doesn't exist. 

I don't see any way a ValueError could be raised.

Instead, just use `.get(key)` on a dictionary, instead of raising it will return `None`, which is what this property is supposed to do when it has no condition data.